frontend/app-development/features/processEditor/handlers/OnProcessTaskRemoveHandler.ts

```diff
@@ -124,5 +124,41 @@ export class OnProcessTaskRemoveHandler {
         layoutSetIdToUpdate: layoutSetId,
       });
     }
+
+    this.removeDeletedSignatureTypeFromTasks(taskMetadata, studioModeler);
+  }
+
+  /**
+   * Removes the deleted signature type from the tasks that are connected to the signature type
+   * @param deletedSigningTask
+   * @param studioModeler
+   * @private
+   */
+  private removeDeletedSignatureTypeFromTasks(
+    deletedSigningTask: OnProcessTaskEvent,
+    studioModeler: StudioModeler,
+  ): void {
+    const signatureDataType =
+      deletedSigningTask.taskEvent.element.businessObject.extensionElements.values[0]
+        .signatureConfig.signatureDataType;
+
+    const tasks = studioModeler.getAllTasksByType('bpmn:Task');
+    const signingTasksToUpdate = tasks.filter(
+      (item) =>
+        item.businessObject.extensionElements.values[0].taskType === 'signing' &&
+        item.businessObject.extensionElements.values[0].signatureConfig?.uniqueFromSignaturesInDataTypes?.dataTypes?.some(
+          (dataType) => dataType.dataType === signatureDataType,
+        ),
+    );
```

Comment:
Just a suggistion:😊
I see that there is a long nested (`.`) which are gathered in long `&&` operator. I tried to destructure `businessObject`, `taskType `and `signatureConfig`, can  we destructure it like this?:
```suggestion
    const tasks = studioModeler.getAllTasksByType('bpmn:Task');
    const signingTasksToUpdate = tasks.filter(({ businessObject: { extensionElements: { values } } }) => {
      const { taskType, signatureConfig } = values[0];
      return taskType === 'signing' &&
        signatureConfig?.uniqueFromSignaturesInDataTypes?.dataTypes?.some(
          ({ dataType }) => dataType === signatureDataType,
        );
    });
```

Reply:
Done!